libs/types/src/core/user/schema/user-activity.schema.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from "zod";
+import { userDtoSchema } from "./user-dto.schema.js";
+import { userEntitySchema } from "./user-entity.schema.js";
+import { activitySchema } from "../../activity/schema/activity.schema.js";
+
+export const userActivitySchema = z.object({
+	userId: userDtoSchema.shape._id,
+	username: userEntitySchema.shape.username,
+	activity: activitySchema
```

Comment:
something weird about this, feel like this causes a compiler issue in the frontend, check it out!

Reply:
or should cause